### PR TITLE
Add documentation for installing and configuring EnMasse

### PR DESCRIPTION
**Pull request by lulf**
I tried to extracts parts of the getting started guide that could fit into the general book. 

I put it into the list of modules at the place I felt it was natural to put it.

### CODE REVIEW

File: documentation/master.adoc [hunk 1]
```diff
@@ -6,8 +6,12 @@ include::modules/address-model.adoc[leveloffset=+1]
 
 include::modules/address-space.adoc[leveloffset=+1]
 
+include::modules/installing-openshift.adoc[leveloffset=+1]
```
Comment (grs) [inline documentation/master.adoc]: My initial instinct is that the we need more than one 'book' here. For the console help, I don't think it makes sense to include the installation guide. Also in terms of sequence, I think for getting started you would want to start with the installation.

Comment (jenmalloy) [reply documentation/master.adoc]: @grs, yes, we should add additional master.adoc files for each guide.

File: documentation/modules/installing-openshift.adoc [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+[source,options="nowrap"]
+----
+curl http://$(oc get route restapi -o jsonpath='{.spec.host}')/v1/addressspaces/default
+----
+
+You can consider the deployment to be finished when 'status.isReady' is 'true' in the returned JSON
+object.
+
+==== Installing in multi-tenant mode
+
```
[the remark this answers is not in the document]
Comment (jenmalloy) [reply documentation/modules/installing-openshift.adoc]: Add ". To install in multi-tenant mode, run the following script:" and add a plus sign on the line above the code example.

File: documentation/modules/installing-openshift.adoc [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+In addition, cluster-admin privileges must be granted
+
+[source,options="nowrap"]
+----
+oc login -u system:admin
+oc adm policy add-cluster-role-to-user cluster-admin system:serviceaccount:$(oc project -q):enmasse-service-account
+oc adm policy add-cluster-role-to-user cluster-admin developer
+----
+
+This will create the deployments required for running {ProductName} in multi-tenant mode.
```
Comment (jenmalloy) [inline documentation/modules/installing-openshift.adoc]: Change to: "The deployments required for running {ProductName} in multi-tenant mode are created." and add a plus sign on line above

File: documentation/modules/installing-openshift.adoc [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+images for the various components.
+
+You can use the REST API to check the status of the deployment.
+
+[source,options="nowrap"]
+----
+curl http://$(oc get route restapi -o jsonpath='{.spec.host}')/v1/addressspaces/default
+----
+
+You can consider the deployment to be finished when 'status.isReady' is 'true' in the returned JSON
```
Comment (jenmalloy) [inline documentation/modules/installing-openshift.adoc]: Change single quote marks to `

File: documentation/modules/installing-openshift.adoc [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+[source,options="nowrap"]
+----
+./deploy-openshift.sh -m "https://localhost:8443" -n enmasse
+----
+
+This will create the deployments required for running {ProductName} in single-tenant mode. Starting
+up {ProductName} will take a while, usually depending on how fast it is able to download the docker
+images for the various components.
+
+You can use the REST API to check the status of the deployment.
```
Comment (jenmalloy) [inline documentation/modules/installing-openshift.adoc]: Make an ordered list item and add a plus sign above the code example below to include it as part of the item. Also change the period to a colon.

File: documentation/modules/installing-openshift.adoc [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+single-tenant mode if you don't expect to be creating multiple isolated address spaces or if you do
+not have cluster-admin privileges. Choose multi-tenant mode otherwise.
+
+==== Installing in single-tenant mode
+
+[source,options="nowrap"]
+----
+./deploy-openshift.sh -m "https://localhost:8443" -n enmasse
+----
+
```
Comment (jenmalloy) [inline documentation/modules/installing-openshift.adoc]: Add "+" to include the paragraph below as part of the item in the ordered list

File: documentation/modules/installing-openshift.adoc [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+up {ProductName} will take a while, usually depending on how fast it is able to download the docker
+images for the various components.
+
+You can use the REST API to check the status of the deployment.
+
+[source,options="nowrap"]
+----
+curl http://$(oc get route restapi -o jsonpath='{.spec.host}')/v1/addressspaces/default
+----
+
```
Comment (jenmalloy) [inline documentation/modules/installing-openshift.adoc]: Add a plus sign above sentence below to include it as part of the list item

**Comment by lulf** [comment]
@jenmalloy Thanks for the thorough review, sorry I didn't notice you've responded that quickly! I've tried to address your comments to the best of my understanding, but please have a look.

**Comment by grs** [comment]
@jenmalloy @lulf I'd suggest we start with two books: a general user guide which would include installation and an online help which would exclude installation and possibly some other bits.

We then need to decide how we want to publish the general user guide both for the latest on master branch and as part of the release process.

**Comment by jenmalloy** [comment]
@grs @lulf Because we were basing the docs on user roles, see my latest PR that includes folders for each book based on user role (and a corresponding master.adoc file for each). Also, the online help parts will be scattered throughout these four guides. Does that make sense?

File: documentation/modules/installing-openshift.adoc [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+Once unpacked, you can deploy {ProductName} using the deployment script provided in the release.
+
+You can invoke the deployment script with `-h` to get a list of options.
+
+{ProductName} can be deployed in either a single-tenant mode or a multi-tenant mode. Choose
+single-tenant mode if you do not expect to be creating multiple isolated address spaces or if you do
+not have cluster-admin privileges. Choose multi-tenant mode otherwise.
+
+==== Installing in single-tenant mode
```
Comment (jenmalloy) [inline documentation/modules/installing-openshift.adoc]: Change to: "Installing {ProductName} in Single-Tenant Mode"

File: documentation/modules/installing-openshift.adoc [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
++
+[source,options="nowrap"]
+----
+curl http://$(oc get route restapi -o jsonpath='{.spec.host}')/v1/addressspaces/default
+----
++
+You can consider the deployment to be finished when `status.isReady` is `true` in the returned JSON
+object.
+
+==== Installing in multi-tenant mode
```
Comment (jenmalloy) [inline documentation/modules/installing-openshift.adoc]: Same as above; add "{ProductName}": "Installing {ProductName} in Multi-Tenant Mode"

Also, add the paragraph from above about whether to choose single-tenant or multi-tenant immediately under this header

File: documentation/modules/installing-openshift.adoc [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+
+.Procedure
+
+. To install in multi-tenant mode, run the following script:
```
Comment (jenmalloy) [inline documentation/modules/installing-openshift.adoc]: Same as above; add "{ProductName}": "To install {ProductName} in..."

File: documentation/modules/installing-openshift.adoc [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
++
+[source,options="nowrap"]
+----
+oc login -u system:admin
+oc adm policy add-cluster-role-to-user cluster-admin system:serviceaccount:$(oc project -q):enmasse-service-account
+oc adm policy add-cluster-role-to-user cluster-admin developer
+----
++
+The deployments required for running {ProductName} in multi-tenant mode are created.
+{ProductName} will be up and running once all pods in the 'enmasse' namespace is in the `Running` state:
```
Comment (jenmalloy) [inline documentation/modules/installing-openshift.adoc]: Change "is" to "are"

File: documentation/master.adoc (hunk 1 above)
Comment (jenmalloy) [reply documentation/master.adoc]: To update my comment above: please see https://github.com/EnMasseProject/enmasse/pull/364
I think that the content below belongs in the new service_admin/master.adoc file?

File: documentation/modules/installing-openshift.adoc [hunk 12]
```diff
@@ -0,0 +1,80 @@
+[[installing-openshift]]
+
+== Installing {ProductName} on OpenShift
+
+[[prerequisites]]
+=== Prerequisites
```
Comment (jenmalloy) [inline documentation/modules/installing-openshift.adoc]: Per the CCS template, change to ". Prerequisites"

File: documentation/modules/installing-openshift.adoc [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+https://github.com/minishift/minishift[minishift] for an example of how to run a local instance of OpenShift
+on your machine.
+
+.Procedure 
+
+. Download one of the releases from https://github.com/EnMasseProject/enmasse/releases and unpack it.
+
+Once unpacked, you can deploy {ProductName} using the deployment script provided in the release.
+
+You can invoke the deployment script with `-h` to get a list of options.
```
Comment (jenmalloy) [inline documentation/modules/installing-openshift.adoc]: Change "get" to "view"

**Comment by lulf** [comment]
@jenmalloy Updated. Agreed on putting it into a separate master.adoc, but should I wait with that until the other PRs are merged?

**Comment by jenmalloy** [comment]
@lulf Yes, if you can wait until the other PR is merged that would be great. I guess I could just merge it now, but let me see if @grs or @rgodfrey have any comments first

**Pull request merged**